lib/widgets/picos_chart_line.dart

```diff
@@ -0,0 +1,127 @@
+
+  List<PicosChartSampleData> _prepareChartData() {
+    if (dailyList == null) return <PicosChartSampleData>[];
+
+    Map<String, DateTime> daysWithDates =
+        PicosChartHelper.getLastSevenDaysWithDates();
+
+    return daysWithDates.entries.map((MapEntry<String, DateTime> entry) {
+      String dayShortText = entry.key;
+      DateTime date = entry.value;
+
+      Daily? matchingDaily = dailyList!.firstWhereOrNull(
+        (Daily daily) => PicosChartHelper.isSameDay(daily.date, date),
+      );
+
+      double? value = matchingDaily?.heartFrequency?.toDouble();
+
+      return PicosChartSampleData(
+        x: dayShortText,
+        y: value,
+      );
+    }).toList();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final GlobalTheme theme = Theme.of(context).extension<GlobalTheme>()!;
+    TooltipBehavior tooltipBehavior = TooltipBehavior(
+      enable: true,
+      textStyle: const TextStyle(
+        fontSize: kIsWeb ? 18 : 14,
+      ),
+    );
+    final DateTime now = DateTime.now();
+    final DateTime sevenDayBefore = now.subtract(const Duration(days: 7));
+    final String titleText =
+        '$title  ${sevenDayBefore.day.toString().padLeft(2, '0')}.'
+        '${sevenDayBefore.month.toString().padLeft(2, '0')}.'
+        '- ${now.day.toString().padLeft(2, '0')}'
+        '.${now.month.toString().padLeft(2, '0')}.${now.year}';
+
+    return Container(
+      decoration: BoxDecoration(
+        border: Border.all(color: theme.blue!, width: PicosChartHelper.width),
+        borderRadius: BorderRadius.circular(10),
+      ),
+      child: SfCartesianChart(
+        plotAreaBorderWidth: 0,
+        title: ChartTitle(
+          text: titleText,
+          textStyle: TextStyle(
+            color: theme.blue,
+            fontWeight: FontWeight.bold,
+            fontSize: 14,
+          ),
+          alignment: ChartAlignment.near,
+        ),
+        primaryXAxis: CategoryAxis(
+          majorGridLines: const MajorGridLines(width: 0),
+          axisLine: AxisLine(
+            color: theme.blue,
+          ),
+          labelStyle: const TextStyle(
+            color: PicosChartHelper.colorBlack,
+          ),
+        ),
+        primaryYAxis: NumericAxis(isVisible: false),
+        tooltipBehavior: tooltipBehavior,
+        series: <LineSeries<PicosChartSampleData, String>>[
+          LineSeries<PicosChartSampleData, String>(
+            dataSource: _prepareChartData(),
+            xValueMapper: (PicosChartSampleData point, _) => point.x,
+            yValueMapper: (PicosChartSampleData point, _) => point.y,
+            name: title,
+            markerSettings: const MarkerSettings(isVisible: true),
+            dataLabelSettings: const DataLabelSettings(
+              isVisible: true,
+              labelAlignment: ChartDataLabelAlignment.top,
+            ),
+            emptyPointSettings: EmptyPointSettings(
+              mode: EmptyPointMode.drop,
+            ),
+          )
```

Comment:
Can you put a trailing comma here?

Reply:
My ide doesn't show me any warnings about this, but if it helps, I'll do it.

Reply:
resolved in [refactoring (trailing comma)](https://github.com/healthcare-it-solutions/picos/pull/184/commits/3e2a7a2a2fe5b0b388ecb61ec59facc80f4a5449)